manager/orchestrator/service.go

```diff
@@ -59,3 +59,41 @@ func DeleteServiceTasks(ctx context.Context, s *store.MemoryStore, service *api.
 		log.G(ctx).WithError(err).Errorf("task search transaction failed")
 	}
 }
+
+// SetServiceTasksRemove sets the desired state of tasks associated with a service
+// to REMOVE, so that they can be properly shut down and later removed by the
+// task reaper.
+func SetServiceTasksRemove(ctx context.Context, s *store.MemoryStore, service *api.Service) {
+	var (
+		tasks []*api.Task
+		err   error
+	)
+	s.View(func(tx store.ReadTx) {
+		tasks, err = store.FindTasks(tx, store.ByServiceID(service.ID))
+	})
```

Comment:
there's a possible race here between the view and the batch below, i think. fix might be to move the view into the batch but before the loop below. not sure if we use this pattern elsewhere.

Reply:
@dperny same pattern as in `DeleteServiceTasks` above this code. I see your point, but does this mean we've missed this race condition in the past?